src/main.rs

```diff
@@ -1,73 +1,83 @@
-#![feature(stdin_forwarders)]
-use crossterm::style::{Color, ResetColor, SetForegroundColor, Stylize};
+use crossterm::style::{Color, Print, ResetColor, SetForegroundColor, Stylize};
 use crossterm::terminal::{Clear, ClearType};
 use crossterm::{cursor, execute, QueueableCommand};
-use lazy_static::lazy_static;
 use rand::Rng;
-use regex::Regex;
 use std::fs;
-use std::io::{self, Write};
+use std::io::{self, BufRead, BufReader, Write};
 
 const FILENAME: &str = "poem.txt";
 const TEXT_WIDTH: u8 = 64;
 
 fn main() {
-    let mut stdin = io::stdin().lines();
+    let mut stdin = BufReader::new(io::stdin()).lines();
     let mut stdout = io::stdout();
 
-    let mut good_answers = 0;
-    let mut bad_answers = 0;
+    let file = fs::File::open(FILENAME).expect("failed loading file");
+    let file = BufReader::new(file);
+    let lines = file
+        .lines()
+        .enumerate()
+        .map(|(n, s)| s.unwrap_or_else(|err| panic!("line {} is invalid: {}", n, err)))
+        .collect::<Vec<_>>();
 
-    let contents = fs::read_to_string(FILENAME).expect(format!("{} not found!", FILENAME).as_str());
-    let lines: Vec<&str> = contents.lines().map(|l| l.trim()).collect();
-
-    for line in lines.iter() {
-        // clear_console(&mut stdout);
-        let (line, hidden_word) = hide_word(line);
-        println!("{}", line.blue());
-        print!("> ");
-        stdout.flush().unwrap();
-        if let Ok(input) = stdin.next().unwrap() {
-            let input = input.trim().to_string();
-
-            if input.to_lowercase() == hidden_word.to_lowercase() {
-                good_answers += 1;
-                stdout.queue(SetForegroundColor(Color::Green)).unwrap();
-                print!("Good!");
+    lines
+        .iter()
+        .enumerate()
+        .fold(0, |correct_answers, (i, line)| {
+            // clear_console(&mut stdout);
+            let (line, hidden_word) = hide_word(line);
+            stdout.queue(Print(line.blue())).unwrap();
+            stdout.queue(Print("> ")).unwrap();
+            stdout.flush().unwrap();
+            let input = stdin.next().unwrap().unwrap();
+            let input = input.trim();
+            let is_valid = input.eq_ignore_ascii_case(&hidden_word);
```

Comment:
`str1.eq_ignore_ascii_case(str2)` is not the same as `str1.to_lowercase() == str2.to_lowercase()` (it only compares ASCII chars)

Reply:
well yes, that's on purpose, some people are too lazy to use diacritics and I wouldn't treat this as a mistake.

Reply:
`eq_ignore_ascii_case` doesn't ignore diacritics